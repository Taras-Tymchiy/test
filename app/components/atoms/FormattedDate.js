// @flow
import * as React from 'react';
import {
    Platform,
    StyleSheet,
    Text
  } from 'react-native';
import moment from 'moment';
import { Post } from '../../entities/Post';

const DEFAULT_FORMAT = 'DD/MM/YYYY HH:mm';

interface DateTimeTextProps {
  value: moment.MomentInput;
  format?: string;
  style?: any
}

export default function FormattedDate({value, format, style}: DateTimeTextProps) {
  const formattedDate = moment(value).format(format || DEFAULT_FORMAT);
  return (
    <Text style={style}>
      {formattedDate}
    </Text>  
  );
};
