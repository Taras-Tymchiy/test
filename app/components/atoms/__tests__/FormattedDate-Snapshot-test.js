import React from 'react';
import renderer from 'react-test-renderer';
import FormattedDate, { FormattedDateProps } from '../FormattedDate';

const props: FormattedDateProps = {
    value: '2015-01-01'
};

it('renders a FormattedDate using Snapshots', () => {
  expect(renderer.create(
    <FormattedDate {...props} />
  )).toMatchSnapshot();
});

it('renders a FormattedDate with custom format using Snapshots', () => {
  const newProps = {
    ...props,
    format: 'YYYY-MM-DD'
  };
  expect(renderer.create(
    <FormattedDate {...newProps} />
  )).toMatchSnapshot();
});

it('renders a FormattedDate with custom styles using Snapshots', () => {
  const newProps = {
    ...props,
    style: {
      color: '#FFF'
    }
  };
  expect(renderer.create(
    <FormattedDate {...newProps} />
  )).toMatchSnapshot();
});