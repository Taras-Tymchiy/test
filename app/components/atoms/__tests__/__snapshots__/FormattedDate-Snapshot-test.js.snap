// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a FormattedDate using Snapshots 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={undefined}
>
  01/01/2015 00:00
</Text>
`;

exports[`renders a FormattedDate with custom format using Snapshots 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={undefined}
>
  2015-01-01
</Text>
`;

exports[`renders a FormattedDate with custom styles using Snapshots 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#FFF",
    }
  }
>
  01/01/2015 00:00
</Text>
`;
